docs/aws/audit/elbmonitoring/rules/fms_webacl_resource_policy_remediation.mdx

```diff
@@ -1,6 +1,223 @@
+3. **Create a WebACL**: Create a WebACL using the WAF client.
+4. **Associate WebACL with ELB**: Associate the created WebACL with your ELB.
+
+By following these steps, you can prevent the misconfiguration where the FMS Policy Owner specifies a WebACLId in an Elastic Load Balancer using Python scripts.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Log in to the AWS Management Console and open the AWS Firewall Manager service.
+
+2. In the Firewall Manager dashboard, select the "Security policies" option from the left-hand side menu.
+
+3. In the Security Policies page, you will see a list of all the policies. Look for the policy that you want to check and click on its name to open its details.
+
+4. In the policy details page, check the "AWS WAF web ACL" field. If the policy owner has specified a WebACLId for the Elastic Load Balancer, it will be displayed here. If this field is empty or not set, then the FMS Policy Owner has not specified a WebACLId.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to execute the commands.
+
+2. Once the AWS CLI is set up, you can use the following command to list all the policies in AWS Firewall Manager (FMS):
+
+   ```
+   aws fms list-policies
+   ```
+   This command will return a list of all FMS policies. Note down the PolicyId of the policy you want to check.
+
+3. Now, use the following command to get the details of the specific policy:
+
+   ```
+   aws fms get-policy --policy-id <PolicyId>
+   ```
+   Replace `<PolicyId>` with the actual PolicyId you noted down in the previous step. This command will return the details of the policy including the WebACLId.
+
+4. Finally, you can use the following command to describe the load balancers and check if the WebACLId is specified in the load balancer:
+
+   ```
+   aws elbv2 describe-load-balancers
+   ```
+   This command will return a list of all the load balancers. Check the WebACLId in the output. If the WebACLId is not specified in the load balancer, then it is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: Before you start, make sure you have the necessary Python libraries installed. You will need the boto3 library, which is the Amazon Web Services (AWS) SDK for Python. It allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, and others. You can install it using pip:
+
+   ```python
+   pip install boto3
+   ```
+
+2. Set up AWS credentials: You need to configure your AWS credentials. You can do this by setting the following environment variables:
+
+   ```python
+   AWS_ACCESS_KEY_ID = 'your_access_key'
+   AWS_SECRET_ACCESS_KEY = 'your_secret_key'
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>